### ISSUE TITLE
LibDogTag-Unit error when turning in quests.

### ISSUE THREAD

**Issue by UppyDan** [ascott18/TellMeWhen#2192]
### WoW Version

Retail

### TellMeWhen Version

11.0.4

### Describe the bug

1x ...ib/LibDogTag-Unit-3.0/Categories/TooltipScanning.lua:204: ...hen/Lib/LibDogTag-Unit-3.0/Categories/Reputation.lua:31: attempt to index local 'data' (a nil value)
[string "=[C]"]: in function `(for generator)'
[string "@TellMeWhen/Lib/LibDogTag-Unit-3.0/Categories/TooltipScanning.lua"]:204: in function `func'
[string "@TellMeWhen/Lib/LibDogTag-3.0-20240726033914/Events.lua"]:301: in function <...rfaceTellMeWhen/Lib/LibDogTag-3.0/Events.lua:294>

Locals:
 = <no value>

This bug happens once in a while after turning in "leveling" quests (quests I am repeating on an alt at the moment).  Apparently GetFactionDataByIndex(i) is returning nil.  Not sure why.  It looks like the code should work fine.

### Export Strings

```text
n/a
```

**Comment by UppyDan** [comment]
ADDITIONAL INFO:
On my character seeing this bug, I did /dump C_Reputation.GetNumFactions() and it returned 26.  I ran a loop using the WowLua addon and each index returned data except for index 8, 9, and 10 which were nil.  Furthermore, I extended the loop up to 50 and it continued to return data beyond the number of factions GetNumFactions() claimed existed.  No other index return nil.  Only 8, 9, and 10 for some reason.

print("START")
local n = C_Reputation.GetNumFactions()
print("# Factions:", n)
for i = 1, n+50 do
   local data = C_Reputation.GetFactionDataByIndex(i)
   print(i, data)
end
